File: java-sdks/src/main/java/com/suisdks/sui/keypairs/Keypair.java
package com.suisdks.sui.keypairs;

import com.suisdks.sui.cryptography.SignatureScheme;

public interface Keypair {
    SignatureScheme scheme();

    byte[] publicKeyBytes();

    byte[] sign(byte[] message);

    boolean verify(byte[] message, byte[] signature);
}

File: kotlin-sdks/src/main/kotlin/com/suisdks/sui/keypairs/Keypair.kt
package com.suisdks.sui.keypairs

import com.suisdks.sui.cryptography.SignatureScheme

interface Keypair {
    fun scheme(): SignatureScheme
    fun publicKeyBytes(): ByteArray
    fun sign(message: ByteArray): ByteArray
    fun verify(message: ByteArray, signature: ByteArray): Boolean
}
